guardian/newsletters-nx
# Define Email Recipients in configuration

## What does this change?

Moves the configuraiton for recipients of email notifications out to the parameter store.

This allows updates to recipents without a code change. 

As the config service is used both in the API and Tool, the config service has been split out to a new shared library. 

The Parameter store has been updated with appropriate config values:

```javascript
{
  "tagRecipients": [
    "newsletters.dev@guardian.co.uk"
  ],
  "brazeRecipients": [
    "newsletters.dev@guardian.co.uk"
  ],
  "signUpPageRecipients": [
    "newsletters.dev@guardian.co.uk"
  ],
  "launchRecipients": [
    "newsletters.dev@guardian.co.uk"
  ],
  "draftCreatedRecipients": [
    "newsletters.dev@guardian.co.uk"
  ]
}
``` 

## How to test

Enable email sending in your local app or on CODE, create, complete & launch an email & verify that emails are sent to the recipients defined in the parameter store:

The parameter store paths are:

- [/DEV/newsletters/newsletters-tool/emailRecipientConfiguration](https://eu-west-1.console.aws.amazon.com/systems-manager/parameters/DEV/newsletters/newsletters-tool/emailRecipientConfiguration/description?region=eu-west-1&tab=Table#list_parameter_filters=Name:Contains:emailRecip)
- [/DEV/newsletters/newsletters-tool/emailRecipientConfiguration](https://eu-west-1.console.aws.amazon.com/systems-manager/parameters/DEV/newsletters/newsletters-tool/emailRecipientConfiguration/description?region=eu-west-1&tab=Table#list_parameter_filters=Name:Contains:emailRecip)
- [/PROD/newsletters/newsletters-tool/emailRecipientConfiguration](https://eu-west-1.console.aws.amazon.com/systems-manager/parameters/PROD/newsletters/newsletters-tool/emailRecipientConfiguration/description?region=eu-west-1&tab=Table#list_parameter_filters=Name:Contains:emailRecip)




## How can we measure success?

Everything should work as before, but now config controlled

## Have we considered potential risks?

Should be fine



## Images

![Screenshot 2023-09-19 at 10 54 12](https://github.com/guardian/newsletters-nx/assets/3277259/1f01b057-e4f9-44e9-9565-7874a26859e0)
